src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +27,24 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        if (!new File(fileName).exists()) {
+            throw new FileNotFoundException("The file doesn`t exist");
+        }
+        String string = Files.readString(Paths.get(fileName));
+        string = string.toLowerCase().replaceAll("[^a-z]", "");
+        Set<Character> set = new TreeSet<>();
+        for (char ch : string.toCharArray()) {
+            set.add(ch);
+        }
+        StringBuilder result = new StringBuilder();
+        List<Character> list = new ArrayList<>(set);
```

Comment:
should we use ArrayList here? try implementing this method without ArrayList

Reply:
You are right. I`ve changed this part of code